api/task_manager_test.cc

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (C) 2022-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#include <seastar/core/coroutine.hh>
+
+#include "task_manager_test.hh"
+#include "api/api-doc/task_manager_test.json.hh"
+#include "task_manager/test_task.hh"
+
+
+namespace api {
+
+namespace tmt = httpd::task_manager_test_json;
+using namespace json;
+
+void set_task_manager_test(http_context& ctx, routes& r) {
+    tmt::register_test_module.set(r, [&ctx] (std::unique_ptr<request> req) -> future<json::json_return_type> {
+        co_await ctx.tm.invoke_on_all([] (task_manager& tm) {
+            tm.make_test_module();
+        });
+        co_return json_void();
+    });
+
+    tmt::unregister_test_module.set(r, [&ctx] (std::unique_ptr<request> req) -> future<json::json_return_type> {
+        co_await ctx.tm.invoke_on_all([name = "test"] (task_manager& tm) -> future<> {
+            auto& modules = tm.get_modules();
+            auto it = modules.find(name);
+            if (it != modules.end()) {
+                co_await it->second->stop();
+            }
+        });
+        co_return json_void();
+    });
+
+    tmt::register_test_task.set(r, [&ctx] (std::unique_ptr<request> req) -> future<json::json_return_type> {
+        sharded<task_manager>& tms = ctx.tm;
+        auto it = req->query_parameters.find("task_id");
+        task_id id = it != req->query_parameters.end() ? task_id{utils::UUID{it->second}} : task_id::create_random_id();
+        it = req->query_parameters.find("keyspace");
+        std::string keyspace = it != req->query_parameters.end() ? it->second : "";
+        it = req->query_parameters.find("table");
+        std::string table = it != req->query_parameters.end() ? it->second : "";
+        it = req->query_parameters.find("type");
+        std::string type = it != req->query_parameters.end() ? it->second : "";
+        it = req->query_parameters.find("entity");
+        std::string entity = it != req->query_parameters.end() ? it->second : "";
+        it = req->query_parameters.find("sequence_number");
+        int sequence_number = it != req->query_parameters.end() ? boost::lexical_cast<int>(it->second) : 0;
```

Comment:
If not given, let's dispatch a new one using the test module.
(can be implemented by the task_manager::module base class in fact since compaction could use it too and repair would eventually be convereted to use it too).
I'm not even sure it makes sense to expose it in the api.
A rest_api test can check it's automatically incrementing by one for subsequent tasks